### PR TITLE
Clarified the ways cookies may be ignored.

### PR DESCRIPTION
**Pull request by johnwilander**
This change introduces the notion of a user agent's cookie policy, briefly explains it, and refers to it in how cookies may be ignored and in user controls. Before this change, a few sections indicated that the user agent MUST accept set-cookie headers while later relaxing that with a reference to privacy/security settings and user controls.

### CODE REVIEW

File: draft-ietf-httpbis-rfc6265bis.md [hunk 1]
```diff
@@ -1721,13 +1728,16 @@ cookies are disabled, the user agent MUST NOT include a Cookie header in
...
+{{ignoring-cookies}}).
+
+User agents MAY provide users the option of preventing persistent storage of
 cookies across sessions. When configured thusly, user agents MUST treat all
 received cookies as if the persistent-flag were set to false. Some popular
 user agents expose this functionality via "private browsing" mode
 {{Aggarwal2010}}.
 
-Some user agents provide users with the ability to approve individual writes to
+User agents MAY provide users with the ability to approve individual writes to
```
Comment (mikewest) [inline draft-ietf-httpbis-rfc6265bis.md]: I wonder if it's reasonable to take this out. I don't think we actually want to encourage any browser to ask users to individually approve writes to the cookie jar.

**Comment by mikewest** [comment]
Ping?

**Comment by mikewest** [comment]
@johnwilander: Ping?

**Comment by chlily1** [comment]
@johnwilander We have a related change in mind that covers a lot of this content. Do you mind if we close this and ping you to review the new change?

File: draft-ietf-httpbis-rfc6265bis.md (hunk 1 above)
Comment (chlily1) [reply draft-ietf-httpbis-rfc6265bis.md]: I agree. Removed.

File: draft-ietf-httpbis-rfc6265bis.md [hunk 2]
```diff
@@ -777,10 +777,20 @@ detail that a user agent implementing these requirements precisely can
 interoperate with existing servers (even those that do not conform to the
 well-behaved profile described in {{sane-profile}}).
 
-A user agent could enforce more restrictions than those specified herein (e.g.,
-for the sake of improved security); however, experiments have shown that such
-strictness reduces the likelihood that a user agent will be able to interoperate
-with existing servers.
```
[the remark this answers is not in the document]
Comment (chlily1) [reply draft-ietf-httpbis-rfc6265bis.md:783]: I added some of this back and referenced the added section.

File: draft-ietf-httpbis-rfc6265bis.md [hunk 3]
```diff
@@ -777,10 +777,20 @@ detail that a user agent implementing these requirements precisely can
...
+## Ignoring Set-Cookie Headers {#ignoring-cookies}
```
[the remark this answers is not in the document]
Comment (chlily1) [reply draft-ietf-httpbis-rfc6265bis.md]: I moved the paragraph describing the cookie policy into the privacy considerations section. The rest of it (regarding 100-level status codes vs all others) was moved down a few sections, as I think that still belongs in the user agent requirements sections.

**Review by chlily1 — COMMENTED** [review]
I addressed the review feedback (mostly moving some things around).

@johnwilander @mikewest @englehardt PTAL, thanks!

**Review by mikewest — COMMENTED** [review]
Only one additional comment.

File: draft-ietf-httpbis-rfc6265bis.md [hunk 4]
```diff
@@ -1068,12 +1068,19 @@ Given a ServiceWorkerGlobalScope (`worker`), the following algorithm returns its
 
 1.  Return `worker`'s origin's host's registered domain.
 
+## Ignoring Set-Cookie Headers {#ignoring-cookies}
+
+User agents MAY ignore Set-Cookie headers contained in responses with 100-level
+status codes or based on its cookie policy (see {{cookie-policy}}).
+
+All other Set-Cookie headers, including those in responses with 400- and 500-level
+status codes, MUST be processed.
```
[the remark this answers is not in the document]
Comment (chlily1) [reply draft-ietf-httpbis-rfc6265bis.md]: I changed it to a SHOULD and spelled it out (hopefully) more clearly. PTAL.

**Review by mikewest — APPROVED** [review]
LGTM. Thanks for polishing this up!

**Pull request merged**